scene/resources/animation.cpp

```diff
@@ -5602,13 +5605,81 @@ Variant Animation::add_variant(const Variant &a, const Variant &b) {
 			return (a.operator Transform3D()) * (b.operator Transform3D());
 		}
 		default: {
+			if (a.is_array()) {
+				switch (a.get_type()) {
+					case Variant::ARRAY: {
+						Array arr_a = a;
+						Array arr_b = b;
+						int size = arr_a.size();
+						if (size == 0 || arr_b.size() != size) {
+							return a;
+						}
+
+						Array result;
+
+						if (arr_a.is_typed() && arr_b.is_typed()) {
```

Comment:
```cpp
if (!arr_a.is_typed() || !arr_b.is_typed()) {
	return a;
}
```
is maybe better.

Reply:
What would the suggestion do?

Reply:
It just recommends an earlier return for non-typed arrays.

Reply:
The idea was to allow interpolation of non-typed Arrays, so long as `a[i]` and `b[i]` are of the same type

Reply:
So what happens when `a[i]` and `b[i]` have different type? Are they skipped? Is there an error?
I think it's ok to support untyped arrays if it doesn't affect performance of the typed ones.

Reply:
The current behavior of this PR is that, if `a[i]` and b[i] are different types they are skipped altogether.

However, as I come back to this PR, I realise that these methods are also generally used by AnimationPlayer. So, I'm not sure if the current implementation break or introduce some odd and unnecessary code smells, right now.